pydocstringformatter/_formatting/formatters_numpydoc.py

```diff
@@ -0,0 +1,135 @@
+from __future__ import annotations
+
+from collections import OrderedDict
+
+from pydocstringformatter._formatting.base import NumpydocSectionFormatter
+
+
+class NumpydocSectionOrderingFormatter(NumpydocSectionFormatter):
+    """Change section order to match numpydoc guidelines."""
+
+    name = "numpydoc-section-order"
+
+    numpydoc_section_order = (
+        # We skip summary, deprecation warning, and extended
+        # summary
+        "Parameters",
+        "Attributes",
+        "Methods",
+        "Returns",
+        "Yields",
+        "Receives",
+        "Other Parameters",
+        "Raises",
+        "Warns",
+        "Warnings",
+        "See Also",
+        "Notes",
+        "References",
+        "Examples",
+    )
+
+    def treat_sections(
+        self, sections: OrderedDict[str, list[str]]
+    ) -> OrderedDict[str, list[str]]:
+        """Sort the numpydoc sections into the numpydoc order."""
+        new_sections = OrderedDict([sections.popitem(last=False)])
+        new_sections.update(
+            OrderedDict(
+                [
+                    (sec_name, sections[sec_name])
+                    for sec_name in sorted(
+                        sections.keys(),
+                        key=self.numpydoc_section_order.index,
+                    )
+                ]
+            )
+        )
+        return new_sections
```

Comment:
Another option, using `OrderedDict` features and avoiding the `KeyError` on weird sections:
```suggestion
        new_sections = sections.copy()
        for sec_name in self.numpydoc_section_order:
            try:
                new_sections.move_to_end(sec_name)
            except KeyError:
                pass
        for sec_name in new_sections.keys()[1:]:
            if sec_name not in self.numpydoc_section_order:
                new_sections.move_to_end(sec_name)
        return new_sections
```

Reply:
Too bad there isn't a `move_to_front`.

Reply:
There's `move_to_back(key, last=False)`, which should do the same thing.  My main reason for ordering it this way is now moot, so I should be able to iterate through `reversed(self.numpydoc_section_order)` without a problem.